### PR TITLE
Simple Testing for matrix_client

### PR DESCRIPTION
**Pull request by Half-Shot**
Figured it would be nice to do some basic testing before a release, so here are a bunch of simple unit tests.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -4,6 +4,8 @@ envlist = py27,py34,py35,packaging,pep8
 [testenv]
 setenv =
     PYTHONDONTWRITEBYTECODE = no_byte_code
+    PYTHONPATH = {toxinidir}/matrix_client
+    TESTING = "unit
```
Comment (SShrike) [inline tox.ini]: Unclosed string.

File: matrix_client/client.py [hunk 2]
```diff
@@ -514,7 +514,7 @@ class User(object):
     """
     def __init__(self, api, user_id):
         if not (user_id.startswith("@") and user_id.count(":") == 1):
-            raise ValueError("UserID's start with @ and seperate with :")
+            raise ValueError("UserID's start with @ and separated with :")
```
Comment (SShrike) [inline matrix_client/client.py]: Should be along the lines of "UserIDs start with '@' and are separated  with ':'"

File: matrix_client/client.py [hunk 3]
```diff
@@ -329,7 +329,7 @@ def __init__(self, client, room_id):
             NOTE: This does not verify the room with the Home Server.
         """
         if not (room_id.startswith("!") and room_id.count(":") == 1):
-            raise ValueError("RoomID's start with ! and seperate with :")
+            raise ValueError("RoomID's start with ! and separated with :")
```
Comment (SShrike) [inline matrix_client/client.py]: Should be along the lines of "Room IDs start with '!' and are separated with ':'".

File: matrix_client/client.py (hunk 2 above)
Comment (Half-Shot) [inline matrix_client/client.py]: > Although the message still remains ambiguous as to what is being separated with ':', but adding extra information would possibly make it too long winded.

Yes, which was my problem. Either I separate the conditions into separate exceptions or I hope the user has a vague idea what they are doing.

Comment (SShrike) [inline matrix_client/client.py]: Yes indeed. In any case, both error messages are grammatically incorrect, so I'd recommend fixing them.

**Comment by Half-Shot** [comment]
After discussions yesterday, it would be better to have a proper wrapper around HTTP requests so we can do some beter testing on api functions.

Hooowever I don't want to get into that now and complicate this release any further. None of the tests in this PR depend on that, so it should all be good?

**Comment by erikjohnston** [comment]
LGTM

**Pull request merged**